src/crafting.cpp

```diff
@@ -1746,219 +1791,277 @@ void set_item_inventory(item &newit)
     }
 }
 
+/* selection of component if a recipe requirement has multiple options (e.g. 'duct tap' or 'welder') */
+comp_selection player::select_component(const std::vector<item_comp> &components, int batch, bool can_cancel) {
+	std::vector<item_comp> player_has;
+	std::vector<item_comp> map_has;
+	std::vector<item_comp> mixed;
+
+	comp_selection selected = comp_selection();
```

Comment:
Well wouldn't ```comp_selection selected;``` leave the values of the struct uninitialized? Isn't it safer to initialize them?